### PR TITLE
GLTFLoader: pbrSpecularGlossiness ext add mipLevel setting

### LINKED ISSUE #15195: GLTFLoader: metal/rough and spec/gloss materials inconsistent.
Per https://github.com/KhronosGroup/glTF-Sample-Models/pull/192, our implementation of the metal/rough and spec/gloss PBR materials is somewhat inconsistent —

47862952-976ff880-de39-11e8-98ed-d327c9eda53e

In theory these two models should match. Currently most WebGL implementations do not have the spec/gloss material correct, but the BabylonJS rendering (https://github.com/KhronosGroup/glTF-Sample-Models/pull/192#issuecomment-419227265) looks like a good reference.

/cc @takahirox this looks similar to the roughness bug you fixed previously?

### PR DESCRIPTION
**Pull request by 06wj**
Fix #15195

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1937,7 +1937,7 @@ function WebGLRenderer( parameters ) {
 			//  WebGLRenderTargetCube will be flipped for backwards compatibility
 			//  WebGLRenderTargetCube.texture will be flipped because it's a Texture and NOT a CubeTexture
 			// this check must be handled differently, or removed entirely, if WebGLRenderTargetCube uses a CubeTexture in the future
-			uniforms.flipEnvMap.value = ( ! ( material.envMap && material.envMap.isCubeTexture ) ) ? 1 : - 1;
+			uniforms.flipEnvMap.value = material.envMap.isCubeTexture ? - 1 : 1;
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: Is this change intentional?

Comment (06wj) [reply src/renderers/WebGLRenderer.js]: Yes, just to reduce the unnecessary calculations.

Comment (mrdoob) [reply src/renderers/WebGLRenderer.js]: What if `material.envMap` is `null`?

Comment (mrdoob) [reply src/renderers/WebGLRenderer.js]: Oh. True that. Thanks!

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -949,11 +949,22 @@ THREE.GLTFLoader = ( function () {
 
 				}
 
-				uniforms.envMap.value = material.envMap;
-				uniforms.envMapIntensity.value = material.envMapIntensity;
-				uniforms.flipEnvMap.value = ( material.envMap && material.envMap.isCubeTexture ) ? - 1 : 1;
+				if (material.envMap) {
```
Comment (mrdoob) [inline examples/js/loaders/GLTFLoader.js]: Minor:

```js
if ( material.envMap ) {
```

**Review by donmccurdy — APPROVED** [review]
| Before | After |
|---|---|
| screen shot 2018-11-17 at 11 30 13 pm | screen shot 2018-11-17 at 11 30 01 pm |

**Comment by mrdoob** [comment]
Thanks!
